http-filter-example/http_filter.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <string>
+
+#include "server/config/network/http_connection_manager.h"
+
+namespace Envoy {
+namespace Http {
+
+class HttpSampleDecoderFilter : public StreamDecoderFilter {
+public:
+  HttpSampleDecoderFilter();
+  ~HttpSampleDecoderFilter();
+
+  // Http::StreamFilterBase
+  void onDestroy() override;
+
+  // Http::StreamDecoderFilter
+  FilterHeadersStatus decodeHeaders(HeaderMap& headers, bool) override;
+  FilterDataStatus decodeData(Buffer::Instance&, bool) override;
+  FilterTrailersStatus decodeTrailers(HeaderMap&) override;
+  void setDecoderFilterCallbacks(StreamDecoderFilterCallbacks& callbacks) override;
+
+private:
+  StreamDecoderFilterCallbacks* decoder_callbacks_;
+
+  static const LowerCaseString headerKey;
```

Comment:
We disallow static const of non-POD in Envoy in the style guide. See the relevant point in https://github.com/lyft/envoy/blob/master/STYLE.md#deviations-from-google-c-style-guidelines for why and how to work around this.